### PR TITLE
Adding initial test script for pipeline

### PR DESCRIPTION
**Pull request by prietyc123**
Get pull_requets and push events ci https://github.com/openshift/release/pull/49986 
And consolidate the base test for sample pipeline into the script

### CODE REVIEW

File: test/validate-pipeline.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+ROOT_DIR=$SCRIPT_DIR/..
+DEBUG_OUTPUT=${DEBUG_OUTPUT:-'/tmp/log.txt'}
+EVENT_TYPE=${EVENT_TYPE:-'push'}
+PIPELINE="${PIPELINE:-docker-build-rhtap}"
+TASK="${TASK:-update-deployment}"
```
Comment (mmorhun) [inline test/validate-pipeline.sh:11]: The same should apply for `deploy-check` task.
I think we should have a list of tasks to check.

File: test/validate-pipeline.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        exit 1
+    fi
+else
+    if oc get taskruns/"$PIPELINE"-"$TASK" -n "$NAMESPACE" -o jsonpath='{.status.conditions[*].reason}' != "Succeeded"; then
+        echo "Error: Gitops deployment repo not updated on push events"
+        exit 1
+    fi
+fi
+
+exit "${status:-0}"
```
Comment (mmorhun) [inline test/validate-pipeline.sh:53]: missing new line

File: test/validate-pipeline.sh [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+echo "Verify the deployment taskruns on push or pull_request event type..."
+if [[ "${EVENT_TYPE}" == "pull_request" ]]; then
+    taskruns=$(oc get pipelineruns/"$PIPELINE" -n "$NAMESPACE" -o jsonpath='{.status.skippedTasks[*].name}')
+    if ! grep -q "$TASK" <<< "$taskruns"; then
+        echo "Error: Gitops deployment repo gets updated on pull_request events"
+        exit 1
+    fi
+else
+    if oc get taskruns/"$PIPELINE"-"$TASK" -n "$NAMESPACE" -o jsonpath='{.status.conditions[*].reason}' != "Succeeded"; then
+        echo "Error: Gitops deployment repo not updated on push events"
```
Comment (mmorhun) [inline test/validate-pipeline.sh:48]: It would show correct message only for `update-deployment` task, but we have `$TASK` as a parameter.

**Review by mmorhun — COMMENTED** [review]
Is `/test` common location for such scripts?

**Comment by prietyc123** [comment]
> Is `/test` common location for such scripts?

Yes I have started thinking of that. For all the test related stuff I would like to put into this folder

**Comment by prietyc123** [comment]
@mmorhun addressed your review comment https://github.com/redhat-appstudio/tssc-sample-pipelines/pull/26/commits/385d163bf6ae0fba4c1e69601dd85872d59e57ee

File: test/validate-pipeline.sh [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    for task in "${TASKS[@]}"; do
+        if ! grep -q "$task" <<< "$taskruns"; then
+            echo "Error: Gitops deployment repo gets modified by task: $task on pull_request events"
+            exit 1
+        fi
+    done
+else
+    for task in "${TASKS[@]}"; do
+        if oc get taskruns/"$PIPELINE"-"$task" -n "$NAMESPACE" -o jsonpath='{.status.conditions[*].reason}' != "Succeeded"; then
+            echo "Error: Gitops deployment repo not modified on push events by task: $task"
```
[the remark this answers is not in the document]
Comment (mmorhun) [reply test/validate-pipeline.sh:52]: You can just have a generic message that mentions the name of the task.

[the remark this answers is not in the document]
Comment (prietyc123) [reply test/validate-pipeline.sh]: ok I have change this to https://github.com/redhat-appstudio/tssc-sample-pipelines/pull/26/files#diff-5fee22e7689e2ba53b6dfe4256ee71b76d7a74efc328aa8f58212391998e14a4R45-R52

**Review by mmorhun — APPROVED** [review]

**Comment by prietyc123** [comment]
@mmorhun I think there is something wrong with the pipeline 

```
Error from server (BadRequest): error when creating "/go/src/github.com/redhat-appstudio/tssc-sample-pipelines/pac/pipelines/gitops-pull-request-rhtap.yaml": admission webhook "validation.webhook.pipeline.tekton.dev" denied the request: validation failed: invalid value: pipeline task "clone-repository" 
expects workspace with name "git-auth" but none exists in pipeline spec: spec.tasks[0].workspaces[1]
invalid value: pipeline task "clone-repository" expects workspace with name "workspace" 
but none exists in pipeline spec: spec.tasks[0].workspaces[0]
invalid value: pipeline task "gather-deploy-images" expects workspace with name "workspace" 
but none exists in pipeline spec: spec.tasks[1].workspaces[0]
```

**Comment by jduimovich** [comment]
> @mmorhun I think there is something wrong with the pipeline
> 
> ```
> Error from server (BadRequest): error when creating "/go/src/github.com/redhat-appstudio/tssc-sample-pipelines/pac/pipelines/gitops-pull-request-rhtap.yaml": admission webhook "validation.webhook.pipeline.tekton.dev" denied the request: validation failed: invalid value: pipeline task "clone-repository" 
> expects workspace with name "git-auth" but none exists in pipeline spec: spec.tasks[0].workspaces[1]
> invalid value: pipeline task "clone-repository" expects workspace with name "workspace" 
> but none exists in pipeline spec: spec.tasks[0].workspaces[0]
> invalid value: pipeline task "gather-deploy-images" expects workspace with name "workspace" 
> but none exists in pipeline spec: spec.tasks[1].workspaces[0]
> ```

The pipeline works as run from the pipelinerun[]( https://github.com/redhat-appstudio/tssc-sample-pipelines/blob/main/pac/gitops-repo/gitops-on-pull-request.yaml) as it defines the workspace and auth. 
Perhaps we can change the tests to use tkn-pac as the resolver for the images which the mirrors how it runs in RHTAP via PaC 

We can also fix the pipeline def to allow `oc apply` use

**Comment by prietyc123** [comment]
This should fix the CI issue https://github.com/openshift/release/pull/50089

/test validate-rhtap-sample-pipelines

**Pull request merged**